lib/inspec/resources/virtualization.rb

```diff
@@ -253,7 +304,40 @@ def collect_data_linux
       return if detect_openvz
       return if detect_openstack
       return if detect_parallels
-      # TODO: where is vmware?
+      return if detect_vmware
+      return if detect_hyperv
+    end
+
+    def windows_computer_system
+      script = "Get-WmiObject -Class Win32_ComputerSystem -ErrorAction Stop | ConvertTo-Json"
+      cmd = inspec.powershell(script)
```

Comment:
It's possible to raise ResourceFailed here with the contents of `cmd.stderr` if there is any error output.